packages/schemas/tables/passcodes.sql

```diff
@@ -0,0 +1,14 @@
+create type passcode_type as enum ('SignIn', 'Register', 'ForgotPassword');
+
+create table passcodes (
+  id varchar(128) not null,
+  session_id varchar(128) not null,
+  phone varchar(32),
+  email varchar(128),
```

Comment:
Maybe a type (`email`| `phone`) and a `target` field would be better?
Is it possible to add new ways for users to receive a passcode? e.g. a code generated by an app, like Microsoft Authenticator

Reply:
1. good point, we may have other ways (like push)
2. MS Authenticator is time-based OTP, which does not need to generate passcode from Logto

Reply:
a session may have both `phone` and `email` passcode, any ideas? @logto-io/eng

Reply:
besides would phone || email be required? At least one must provided.

Reply:
Yes, but is hard to add this constraint in DB

Reply:
I was thinking a field like `indicator not null` could be either phone and email or any. and another field indictor_type: enum. phone , email sth like that.

Reply:
feel free to resolve this one, if we haven;t come up with a better solution.